### PR TITLE
Add MMMC corners json and setup scripts

### PR DESCRIPTION
**Pull request by colinschmidt**
This seems to work for my use case.

### CODE REVIEW

File: src/hammer-vlsi/hammer_vlsi.py [hunk 1]
```diff
@@ -486,6 +486,29 @@ def from_string(s: str) -> "PlacementConstraintType":
...
+        elif s == "hold":
+            return MMMCCornerType.Hold
+        elif s == "extra":
+            return MMMCCornerType.Extra
+        else:
+            raise ValueError("Invalid mmmc corner type '{}'".format(s))
+
+MMMCCorner = NamedTuple('MMMCCorner', [
+    ('name', str),
+    ('type', PlacementConstraintType),
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi.py]: Typo - MMMCCornerType?

File: src/hammer-vlsi/hammer_vlsi.py [hunk 2]
```diff
@@ -2187,16 +2227,110 @@ def append_mmmc(cmd: str) -> None:
...
+        if(corners):
+            setup_corner = corners[0]
+            hold_corner = corners[0]
+            # TODO (colins): handle more than one corner and do something with extra corners
+            for corner in corners:
+                if(corner.type is MMMCCornerType.Setup):
+                    setup_corner = corner
+                if(corner.type is MMMCCornerType.Hold):
+                    hold_corner = corner
+            print("hold:")
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi.py]: Leftover debug statements?

File: src/hammer-vlsi/hammer_vlsi.py [hunk 3]
```diff
@@ -2187,16 +2227,110 @@ def append_mmmc(cmd: str) -> None:
...
-        append_mmmc("create_analysis_view -name {name} -delay_corner {corner} -constraint_mode {constraint}".format(
-            name=analysis_view_name, corner=delay_corner_name, constraint=constraint_mode))
-        # Finally, apply the analysis view.
-        # TODO: introduce different views of setup/hold and true multi-corner
-        append_mmmc("set_analysis_view -setup {{ {setup_view} }} -hold {{ {hold_view} }}".format(
-            setup_view=analysis_view_name,
-            hold_view=analysis_view_name
-        ))
+        corners = self.get_mmmc_corners()
+        print(corners)
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi.py]: Debug statements?

**Comment by colinschmidt** [comment]
Thanks for the fixes

**Pull request merged**